front/src/applications/stdcmV2/components/StdcmOrigin.tsx

```diff
@@ -31,12 +32,20 @@ const StdcmOrigin = ({ setCurrentSimulationInputs, disabled = false }: StdcmConf
   useEffect(() => {
     setCurrentSimulationInputs((prevState) => ({
       ...prevState,
-      origin,
+      pathSteps: replaceElementAtIndex(prevState?.pathSteps, 0, origin),
       departureDate: originDate,
       departureTime: originTime,
     }));
   }, [origin, originDate, originTime]);
 
+  const onOriginDateInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(updateOriginDate(e.target.value));
+  };
+
+  const onOriginTimeInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(updateOriginTime(e.target.value));
+  };
+
```

Comment:
Maybe we can insert them directly in the `onChange` ?

Reply:
I chose to do it this way to prepare for component testing. When the tests are implemented, these methods will be provided via a hook, so that the component is really a visual, dumb component devoid of logic and only declarative.
This hook will then be tested.

Reply:
Ok understood ! Should these be wrapped with a `useCallBack` then ?

Reply:
Yes, that's right! As recommended in the [react doc.](Yes, that's right! As recommended in the react doc.)